spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDParser.java

```diff
@@ -43,4 +45,29 @@
 	 *         described by the DSL.
 	 */
 	List<ModuleDescriptor> parse(String name, String config, ParsingContext type);
+
+	public static class DefinitionOrException {
```

Comment:
It seem like the loop should move up to the FloController then. If that's the only place it's used, we don't want to add complexity at this level.